File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    var answer = 0;
    const visited = Array.from({length: n}, ()=> false);
    
    
    function dfs(idx){
        visited[idx] = true;
        
        for(let i=0; i < computers[idx].length; i++){
            if(!visited[i] && computers[idx][i]===1){
                dfs(i);
            }
        }
    }
    
    // Node 차례대로 start!
    for(let i=0; i<n; i++){
        if(!visited[i]){
            answer++;
            dfs(i);
        }
    }
    
    return answer;
}

File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False] * n
    
    def dfs(node):
        visited[node] = True
        
        for computer in computers[node]:
            for c in range(n):
                if not visited[c] and computers[node][c] == 1:
                    dfs(c)
         
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(i)
        
    
    return answer